arch/arm/soc/st_stm32/stm32l0/soc_gpio.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2016 Open-RnD Sp. z o.o.
+ * Copyright (c) 2016 BayLibre, SAS
+ * Copyright (c) 2018 Endre Karlson <endre.karlson@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief
+ *
+ * Based on reference manual:
+ *   STM32L0X advanced ARM ® -based 32-bit MCUs
+ *
+ * General-purpose I/Os (GPIOs)
+ */
+
+#include <errno.h>
+
+#include <device.h>
+#include "soc.h"
+#include "soc_registers.h"
+#include <gpio.h>
+#include <gpio/gpio_stm32.h>
+
+
+enum {
+	STM32L0X_PIN3			= 3,
+	STM32L0X_PIN7			= 7,
+	STM32L0X_PIN11			= 11,
+	STM32L0X_PIN15			= 15,
+};
+
+#define STM32L0X_IDR_PIN_MASK		0x1
+#define STM32L0X_AFR_MASK		0xf
+
+/* GPIO registers - each GPIO port controls 16 pins */
+struct stm32l0x_gpio {
+	u32_t moder;
+	u32_t otyper;
+	u32_t ospeedr;
+	u32_t pupdr;
+	u32_t idr;
+	u32_t odr;
+	u32_t bsrr;
+	u32_t lckr;
+	u32_t afr[2];
+	u32_t brr;
+	u32_t ascr; /* Only present on STM32L0x1, STM32L0x5, STM32L0x6 */
```

Comment:
There is no STM32L0x5/6, and, AFAIK, there is no such register in STM32L0.

Reply:
I removed it.